SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,255 @@
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("2")]
+		public IActionResult SecondStep(SecondStepModel model)
+		{
+			var countries = new CountryHelper().GetCountryData();
+
+			if (ModelState.IsValid)
+			{
+				var country = countries.Where(c => c.CountryShortCode == model.CountryCode);
+				if (country.Any())
+				{
+					GlobalSettings.Store("CONFIG_CompanyName", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanySubject", model.CompanyName);
+					GlobalSettings.Store("CONFIG_CompanyCountryCode", country.First().CountryShortCode);
+					GlobalSettings.Store("CONFIG_URL", model.AppHost);
+
+					Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "3");
+					GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "3");
+
+					return Redirect("3");
+				}
+				else
+				{
+					ModelState.AddModelError(string.Empty, "Unknown country");
+				}
+
+			}
+
+			model.CountryList = new Microsoft.AspNetCore.Mvc.Rendering.SelectList(countries, "CountryShortCode", "CountryName");
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("3")]
+		public IActionResult ThirdStep()
+		{
+			var ldapHost = GlobalSettings.Get("LDAP_Host");
+			var ldapPort = GlobalSettings.GetInt("LDAP_Port");
+			var ldapSSL = GlobalSettings.Get("LDAP_SSL") != null ? bool.Parse(GlobalSettings.Get("LDAP_SSL")) : false;
+			var ldapBindDn = GlobalSettings.Get("LDAP_BindDN");
+			var ldapBindPassword = GlobalSettings.Get("LDAP_BindPass");
+
+			ThirdStepModel model = new ThirdStepModel();
+			model.Host = ldapHost != null ? ldapHost : "";
+			model.Port = ldapPort != null ? ((int)ldapPort) : 389;
+			model.SSL = ldapSSL;
+			model.BindDN = ldapBindDn != null ? ldapBindDn : "";
+			model.BindPassword = ldapBindPassword != null ? ldapBindPassword : "";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("3")]
+		public IActionResult ThirdStep(ThirdStepModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				LdapConnection connection = new LdapConnection();
+				connection.SecureSocketLayer = model.SSL;
+
+				try
+				{
+					connection.Connect(model.Host, model.Port);
+					connection.Bind(model.BindDN, model.BindPassword);
+
+					//If everything goes well
+					GlobalSettings.Store("LDAP_Host", model.Host);
+					GlobalSettings.Store("LDAP_Port", model.Port.ToString());
+					GlobalSettings.Store("LDAP_SSL", model.SSL.ToString());
+					GlobalSettings.Store("LDAP_BindDN", model.BindDN);
+					GlobalSettings.Store("LDAP_BindPass", model.BindPassword);
+
+					Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "4");
+					GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "4");
+					return Redirect("4");
+
+				}
+				catch (LdapException ex)
+				{
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (AggregateException ex)
+				{
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (Exception ex)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Catch a list of specific exception subtype or use exception filters instead.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)